src/codegen/polkadot.rs

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains `codegen` helpers for the Polkadot target.
+
+use solang_parser::pt::Loc;
+
+use crate::{
+    codegen::{
+        cfg::{ControlFlowGraph, Instr},
+        revert::log_runtime_error,
+        vartable::Vartable,
+        Expression, Options,
+    },
+    sema::ast::{Namespace, Type},
+};
+
+/// Helper to handle error cases from external function and constructor calls.
+pub(crate) struct RetCodeCheck {
+    pub success: usize,
+    pub revert: usize,
+    pub error_no_data: usize,
+    msg: &'static str,
+    loc: Loc,
+}
+
+#[derive(Default)]
+pub(crate) struct RetCodeCheckBuilder<Success = ()> {
+    loc: Loc,
+    msg: &'static str,
+    success_var: Success,
+}
+
+impl RetCodeCheckBuilder {
+    pub(crate) fn loc(self, loc: Loc) -> Self {
+        Self {
+            loc,
+            msg: self.msg,
+            success_var: self.success_var,
+        }
+    }
+
+    pub(crate) fn msg(self, msg: &'static str) -> Self {
+        Self {
+            loc: self.loc,
+            msg,
+            success_var: self.success_var,
+        }
```

Comment:
```rust
self.msg = msg;
self
```
Seems less error prone (e.g. if we add a field)?

Reply:
I don't think it matters as it's purely a question of style. I changed it for now (also this was already ✅ before and not related to what I changed since then)